Dockerfile

```diff
@@ -0,0 +1,32 @@
+FROM ubuntu
+
+# Install needed packages.
+RUN apt-get update -q
+RUN apt-get install -qy curl libmysqlclient-dev libmagickwand-dev libmagic-dev cmake imagemagick
+
+# Install ruby
+RUN gpg --keyserver hkp://keys.gnupg.net --recv-keys 409B6B1796C275462A1703113804BB82D39DC0E3
+RUN curl -sSL https://get.rvm.io | bash -s stable
+RUN /bin/bash -c -l 'rvm requirements'
+RUN /bin/bash -c -l 'rvm install ruby-2.2.0'
+RUN /bin/bash -c -l 'rvm use 2.2.0'
+RUN /bin/bash -c -l 'gem install bundler --no-ri --no-rdoc'
+
+
+# Create a directory for the app.
+RUN mkdir -p /glittergallery
+WORKDIR /glittergallery
+
+# Allow reaching via port 3000.
+EXPOSE 3000
+
+# Install required gems.
+ADD Gemfile /glittergallery/Gemfile
```

Comment:
Docker keeps caches after each step, copying the whole thing would make the cache size bigger if it fails at bundling for example. It's advised to keep that part of copying the actual app to the end of the file :)